libraries/GCS_MAVLink/GCS_Signing.cpp

```diff
@@ -143,7 +148,9 @@ void GCS_MAVLINK::load_signing_key(void)
     // prevents a window for replay attacks
     signing.timestamp = key.timestamp + 60UL * 100UL * 1000UL;
     signing.flags = MAVLINK_SIGNING_FLAG_SIGN_OUTGOING;
-    signing.accept_unsigned_callback = accept_unsigned_callback;
+    signing.accept_unsigned_callback = option_enabled(Option::MAVLINK2_SIGNING_DISABLED)
+        ? always_true
+        : accept_unsigned_callback;
```

Comment:
```suggestion
    if (option_enabled(Option::MAVLINK2_SIGNING_DISABLED) || chan == MAVLINK_COMM_0) {
        signing.accept_unsigned_callback = always_true;
    } else {
        signing.accept_unsigned_callback = accept_unsigned_callback;
    }
```

... and remove the `if` in `accept_unsigned_callback` for COMM_0